src/verify_expr.ml

```diff
@@ -384,6 +384,30 @@ module VerifyExpr(VerifyProgramArgs: VERIFY_PROGRAM_ARGS) = struct
     | _ -> false
 
   let reportStmts ss = List.iter (stmt_iter (fun s -> if not (is_transparent_stmt s) then reportStmt (stmt_loc s))) ss
+
+  let check_cxx_spec_overrides fenv (meth_name, meth_info) get_meth_info =
+    let FuncInfo ([], fterm, l, k, tparams, rt, xmap, ng_callers_only, pre, pre_tenv, post, terminates, _, _, is_virtual, overrides) = meth_info in
+    match overrides with
+    | [] -> ()
+    | _ ->
+      let ("this", PtrType (StructType derived)) :: xmap = xmap in
+      let this_term = get_unique_var_symb_non_ghost "this" (PtrType (StructType derived)) in
+      let env = ("this", this_term) :: fenv in
+      let rec check (prev_base_name, prev_base_term) overrides =
+        match overrides with
+        | [] -> ()
+        | override :: overrides ->
+          let FuncInfo ([], fterm0, l0, k0, tparams0, rt0, xmap0, ng_callers_only0, pre0, pre_tenv0, post0, terminates0, _, _, is_virtual0, overrides0) = get_meth_info override in
+          let ("this", PtrType (StructType base)) :: xmap0 = xmap0 in
+          let base_term = direct_base_addr (prev_base_name, prev_base_term) base in
+          let () = check_func_header_compat l ("Method '" ^ meth_name ^ "'") ("Method implementation check of '" ^ override ^ "'") env

```

Comment:
Message: should this not be "override check"? ("Implementation check" sounds like you're checking the method's body.) And should it not mention both `override` and `derived`?